src/NUnitEngine/nunit.engine.tests/Services/RuntimeFrameworkServiceTests.cs

```diff
@@ -92,6 +92,30 @@ public void EngineOptionPreferredOverImageTarget(string framework, int majorVers
             _runtimeService.SelectRuntimeFramework(package);
             Assert.That(package.GetSetting<string>(EnginePackageSettings.RuntimeFramework, null), Is.EqualTo(requested));
         }
+
+        [Test]
+        public void RuntimeFrameworkIsSetForSubpackages()
+        {
+            //Runtime Service verifies that requested frameworks are available, therefore this test can only currently be run on platforms with both CLR v2 and v4 available
+            Assume.That(new RuntimeFramework(RuntimeType.Net, new Version("2.0.50727")), Has.Property(nameof(RuntimeFramework.IsAvailable)).True);
+            Assume.That(new RuntimeFramework(RuntimeType.Net, new Version("4.0.30319")), Has.Property(nameof(RuntimeFramework.IsAvailable)).True);
+
+            var topLevelPackage = new TestPackage(new [] {"a.dll", "b.dll"});
+
+            var net20Package = topLevelPackage.SubPackages[0];
+            net20Package.Settings.Add(InternalEnginePackageSettings.ImageRuntimeVersion, new Version("2.0.50727"));
+            var net40Package = topLevelPackage.SubPackages[1];
+            net40Package.Settings.Add(InternalEnginePackageSettings.ImageRuntimeVersion, new Version("4.0.30319"));
+
+            _runtimeService.SelectRuntimeFramework(topLevelPackage);
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(net20Package.Settings[EnginePackageSettings.RuntimeFramework], Is.EqualTo("net-2.0"));
+                Assert.That(net40Package.Settings[EnginePackageSettings.RuntimeFramework], Is.EqualTo("net-4.0"));
+                Assert.That(topLevelPackage.Settings[EnginePackageSettings.RuntimeFramework], Is.EqualTo("net-4.0"));
+            });
+        }
```

Comment:
This stuff ain't great to test, hence the restriction. Along with the refactor, I'd love us to set up some better mocks for this service!